krun.py

```diff
@@ -140,13 +125,16 @@ class ScheduleEmpty(Exception):
 class ExecutionScheduler(object):
     """Represents our entire benchmarking session"""
 
-    def __init__(self, config_file, log_filename, out_file, mailer, platform):
+    def __init__(self, config_file, log_filename, out_file, mailer, platform,
```

Comment:
I think if we have the class then we need to use it everywhere and not have the dict, rather than having both.